1/1.4/Dockerfile

```
@@ -87,8 +89,9 @@ RUN mkdir -p /tmp/build && \
     unzip -d /bin consul_${CONSUL_VERSION}_linux_amd64.zip && \
     cd /tmp && \
     rm -rf /tmp/build && \
-# tiny smoke test to ensure the binary we downloaded runs
-    consul version
+# tiny smoke test to ensure the binary we downloaded runs right version
+    test ${C2D_RELEASE} = $(consul version | awk '/Consul/ { print $2 }' | grep -o -E '([0-9]+\.)([0-9]+\.)([0-9]+)'$) \
+         || echo "Version check failed"
```

If the test failure then the result is printing the message with `0` exit code.

```suggestion
         || { echo "Version check failed"; exit 1; }
```

---

Thanks for suggestions!

---

Fixed